### PR TITLE
Israel.5490.governance mixpanel

### PR DESCRIPTION
**Pull request by Israellund**
## Link to Issue
Closes: #5490

## Description of Changes
Added Mixpanel Analytics to Snapshot voting and Governance proposals & voting. Changed logic withink onClicks to conform to updated code standards
NOTE: There was already a Snapshot Creation Mixpanel event in the codebase so I did not add to the first checkbox. And I was unable to test on the on-chain proposals, as there is no testnet available for that. 

## Test Plan
NOTE: Only Snapshot creation and votes are able to be tested as there is no testnet for the on-chain proposals. 
-Create a Snapshot
-Confirm in Mixpanel that this occurred
-Cast a vote on a Snapshot
-Confirm in Mixpanel that this occurred.

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/modals/confirm_snapshot_vote_modal.tsx [hunk 1]
```diff
@@ -83,29 +114,7 @@ export const ConfirmSnapshotVoteModal = (
...
-                successCallback();
-              });
-            } catch (err) {
-              console.log(err);
-              const errorMessage = err.message;
-              notifyError(errorMessage);
-            }
-            setIsSaving(false);
-          }}
+          onClick={handleVote}
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/confirm_snapshot_vote_modal.tsx:117]: good stuff!

File: packages/commonwealth/client/scripts/views/modals/confirm_snapshot_vote_modal.tsx [hunk 2]
```diff
@@ -44,6 +45,36 @@ export const ConfirmSnapshotVoteModal = (
...
+    } catch (err) {
+      console.log(err);
+      const errorMessage = err.message;
+      notifyError(errorMessage);
+    }
+    setIsSaving(false);
+
+    trackAnalytics({
+      event: MixpanelSnapshotEvents.SNAPSHOT_VOTE_OCCURRED,
+    });
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/confirm_snapshot_vote_modal.tsx:75]: event will be fired regardless success or failure. Shouldn't it be only at the end of `try` cluase?

File: packages/commonwealth/client/scripts/views/components/proposals/voting_actions.tsx [hunk 3]
```diff
@@ -103,71 +106,115 @@ export const VotingActions = (props: VotingActionsProps) => {
...
-        .catch((err) => notifyError(err.toString()));
+      try {
+        await proposal.voteTx(new CosmosVote(user, 'No'));
+        emitRedraw();
+      } catch (err) {
+        notifyError(err.toString());
+      }
+      trackAnalytics({
+        event: MixpanelGovernanceEvents.COSMOS_VOTE_OCCURRED,
+      });
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/proposals/voting_actions.tsx:183]: same here, shouldn't it be before catch?

File: packages/commonwealth/client/scripts/views/components/proposals/voting_actions.tsx [hunk 4]
```diff
@@ -103,71 +106,115 @@ export const VotingActions = (props: VotingActionsProps) => {
...
+        emitRedraw();
+        trackAnalytics({
+          event: MixpanelGovernanceEvents.AAVE_VOTE_OCCURRED,
+        });
+      } catch (err) {
+        notifyError(err.toString());
+      }
+      trackAnalytics({
+        event: MixpanelGovernanceEvents.SPUTNIK_VOTE_OCCURRED,
+      });
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/proposals/voting_actions.tsx:208]: and here

File: packages/commonwealth/client/scripts/views/pages/new_proposal/sputnik_proposal_form.tsx [hunk 5]
```diff
@@ -81,46 +124,7 @@ export const SputnikProposalForm = () => {
...
-          } else {
-            throw new Error('unsupported sputnik proposal type');
-          }
-
-          (app.chain as NearSputnik).dao
-            .proposeTx(description, propArgs)
-            .catch((err) => notifyError(err.message));
-        }}
-      />
+      <CWButton label="Send transaction" onClick={handleSendTransaction} />
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/sputnik_proposal_form.tsx:127]: thanks for moving all of these inline functions into the handlers! This way we declutter jsx/visual part of the code

File: packages/commonwealth/client/scripts/views/pages/new_proposal/aave_proposal_form.tsx [hunk 6]
```diff
@@ -56,6 +58,71 @@ export const AaveProposalForm = () => {
...
+      values,
+      calldatas,
+      signatures,
+      withDelegateCalls,
+      ipfsHash: _ipfsHash,
+    };
+
+    // aave.governance
+    //   .propose(details)
+    //   .catch((err) => notifyError(err.data?.message || err.message));
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/aave_proposal_form.tsx:114]: what about this? It was not commented out

**Review by masvelio — APPROVED** [review]
didn't test if works with mixpanel. Few small comments to the code but generally looks good 💪

**Comment by Israellund** [comment]
@masvelio Just pushed changes, thanks for catching those. I guess I went a little cross-eyed looking at so many else if and try/catch statements.

File: packages/commonwealth/client/scripts/views/modals/confirm_snapshot_vote_modal.tsx [hunk 7]
```diff
@@ -1,19 +1,20 @@
 import React from 'react';
 
 import type { SnapshotProposal, SnapshotSpace } from 'helpers/snapshot_utils';
-import app from '../../state';
+import { useBrowserAnalyticsTrack } from 'hooks/useBrowserAnalyticsTrack';
+import { formatNumberShort } from '../../../../shared/adapters/currency';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/modals/confirm_snapshot_vote_modal.tsx:5]: Please don't use relative imports for these.

File: packages/commonwealth/client/scripts/views/pages/new_proposal/aave_proposal_form.tsx [hunk 8]
```diff
@@ -1,15 +1,16 @@
 import { utils } from 'ethers';
 import React, { useEffect, useState } from 'react';
 
-import 'pages/new_proposal/aave_proposal_form.scss';
-
+import { useBrowserAnalyticsTrack } from 'client/scripts/hooks/useBrowserAnalyticsTrack';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/aave_proposal_form.tsx:4]: Remove client/scripts prefix.

File: packages/commonwealth/client/scripts/views/pages/new_proposal/aave_proposal_form.tsx [hunk 9]
```diff
@@ -1,15 +1,16 @@
...
 import { notifyError } from 'controllers/app/notifications';
 import type Aave from 'controllers/chain/ethereum/aave/adapter';
 import { AaveExecutor } from 'controllers/chain/ethereum/aave/api';
 import type { AaveProposalArgs } from 'controllers/chain/ethereum/aave/governance';
+import 'pages/new_proposal/aave_proposal_form.scss';
 import app from 'state';
 import { PopoverMenu } from 'views/components/component_kit/CWPopoverMenu';
 import { User } from 'views/components/user/user';
+import { MixpanelGovernanceEvents } from '../../../../../shared/analytics/types';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/aave_proposal_form.tsx:13]: Use absolute import.

File: packages/commonwealth/client/scripts/views/pages/new_proposal/compound_proposal_form.tsx [hunk 10]
```diff
@@ -1,14 +1,13 @@
-import React, { useState } from 'react';
-
+import { useBrowserAnalyticsTrack } from 'client/scripts/hooks/useBrowserAnalyticsTrack';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/compound_proposal_form.tsx:1]: remove client/scripts prefix

File: packages/commonwealth/client/scripts/views/pages/new_proposal/compound_proposal_form.tsx [hunk 11]
```diff
@@ -1,14 +1,13 @@
...
 import type Compound from 'controllers/chain/ethereum/compound/adapter';
 import type { CompoundProposalArgs } from 'controllers/chain/ethereum/compound/governance';
-
 import 'pages/new_proposal/compound_proposal_form.scss';
-
+import React, { useState } from 'react';
 import app from 'state';
 import { PopoverMenu } from 'views/components/component_kit/CWPopoverMenu';
 import { User } from 'views/components/user/user';
+import { MixpanelGovernanceEvents } from '../../../../../shared/analytics/types';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/compound_proposal_form.tsx:10]: use absolute import

File: packages/commonwealth/client/scripts/views/pages/new_proposal/cosmos_proposal_form.tsx [hunk 12]
```diff
@@ -1,6 +1,4 @@
-import type { Any as ProtobufAny } from 'cosmjs-types/google/protobuf/any';
-import React, { useState } from 'react';
-
+import { useBrowserAnalyticsTrack } from 'client/scripts/hooks/useBrowserAnalyticsTrack';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/cosmos_proposal_form.tsx:1]: see above

File: packages/commonwealth/client/scripts/views/pages/new_proposal/cosmos_proposal_form.tsx [hunk 13]
```diff
@@ -9,13 +7,15 @@ import {
...
+import React, { useState } from 'react';
+import app from 'state';
 import {
   useDepositParamsQuery,
   useStakingParamsQuery,
 } from 'state/api/chainParams';
-
-import { useCommonNavigate } from 'navigation/helpers';
-import app from 'state';
+import { MixpanelGovernanceEvents } from '../../../../../shared/analytics/types';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/cosmos_proposal_form.tsx:18]: see above

File: packages/commonwealth/client/scripts/views/pages/new_proposal/sputnik_proposal_form.tsx [hunk 14]
```diff
@@ -1,10 +1,11 @@
 import React, { useState } from 'react';
 
+import { useBrowserAnalyticsTrack } from 'client/scripts/hooks/useBrowserAnalyticsTrack';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/sputnik_proposal_form.tsx:3]: see above

File: packages/commonwealth/client/scripts/views/pages/new_proposal/sputnik_proposal_form.tsx [hunk 15]
```diff
@@ -1,10 +1,11 @@
...
 import { notifyError } from 'controllers/app/notifications';
 import type NearSputnik from 'controllers/chain/near/sputnik/adapter';
 import type { NearSputnikProposalKind } from 'controllers/chain/near/sputnik/types';
-
 import app from 'state';
+import { MixpanelGovernanceEvents } from '../../../../../shared/analytics/types';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/new_proposal/sputnik_proposal_form.tsx:8]: see above

File: packages/commonwealth/client/scripts/views/pages/view_snapshot_proposal/snapshot_poll_card.tsx [hunk 16]
```diff
@@ -1,6 +1,7 @@
-import React, { useEffect } from 'react';
-
+import { useBrowserAnalyticsTrack } from 'client/scripts/hooks/useBrowserAnalyticsTrack';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/view_snapshot_proposal/snapshot_poll_card.tsx:1]: see above

File: packages/commonwealth/client/scripts/views/pages/view_snapshot_proposal/snapshot_poll_card.tsx [hunk 17]
```diff
@@ -1,6 +1,7 @@
...
 import 'components/poll_card.scss';
+import React, { useEffect } from 'react';
+import { MixpanelSnapshotEvents } from '../../../../../shared/analytics/types';
```
Comment (jnaviask) [inline packages/commonwealth/client/scripts/views/pages/view_snapshot_proposal/snapshot_poll_card.tsx:4]: see above

**Review by jnaviask — CHANGES_REQUESTED** [review]
Any way to at least confirm these pages render correctly? What communities did you use?

File: packages/commonwealth/client/scripts/views/components/ProposalCard/ProposalCard.tsx [hunk 18]
```diff
@@ -48,15 +47,6 @@ export const ProposalCard = ({
     }
   }, [proposal]);
 
-  useEffect(() => {
```
[the remark this answers is not in the document]
Comment (Israellund) [reply packages/commonwealth/client/scripts/views/components/ProposalCard/ProposalCard.tsx:51]: @jnaviask There were duplicate useEffects.
https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/client/scripts/views/components/ProposalCard/ProposalCard.tsx
Screenshot 2024-02-07 at 9 52 05 PM

**Comment by Israellund** [comment]
I commented out all checks and `governance.proposal()` call in `aave_proposal_form.tsx` to check that the event shows up on the Mixpanel dashboard. See video.

https://github.com/hicommonwealth/commonwealth/assets/69872984/0fc17266-495b-4478-9622-00d2ae10beb6

**Comment by CowMuon** [comment]
Is this ready for review again? @Israellund (in any case, should go into 0.9.6 this week)

**Comment by Israellund** [comment]
@CowMuon yes it should be good to review again

**Review by Miaplacidus — APPROVED** [review]
LGTM

**Pull request merged**